roles/ceph-facts/tasks/facts.yml

```diff
@@ -302,7 +302,16 @@
         chrony_daemon_name: chrony
       when: ansible_os_family == "Debian"
 
-- name: set grafana_server_addr fact
+- name: set grafana_server_addr fact - ipv4
   set_fact:
     grafana_server_addr: "{{ (hostvars[groups[grafana_server_group_name][0] | default(groups[mgr_group_name][0])])['ansible_all_ipv4_addresses'] | ipaddr(public_network) | first }}"
-  when:  groups.get(grafana_server_group_name, []) | length > 0 or groups.get(mgr_group_name, []) | length > 0
+  when:
+    - groups.get(grafana_server_group_name, []) | length > 0 or groups.get(mgr_group_name, []) | length > 0
+    - ip_version == 'ipv4'
+
+- name: set grafana_server_addr fact - ipv6
+  set_fact:
+    grafana_server_addr: "{{ (hostvars[groups[grafana_server_group_name][0] | default(groups[mgr_group_name][0])])['ansible_all_ipv6_addresses'] | ipaddr(public_network) | first }}"
```

Comment:
I don't know if it's the best place here or where the `grafana_server_addr` variable is used but this needs to use brackets for ipv6 by adding the `ipwrap` filter.

https://github.com/ceph/ceph-ansible/blob/master/roles/ceph-facts/tasks/set_monitor_address.yml#L14

Also the first filter should be replace by last with ipv6

Reply:
Thanks for the suggestion, I made the requested changes.
This variable is used in many templates (grafana datasource, prometheus, alertmanager), for this reason I think it's good setting a fact that could be consumed from many roles.